pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -121,6 +125,24 @@ func validateIPAddresses(a *cmapi.CertificateSpec, fldPath *field.Path) field.Er
 	return el
 }
 
+func validateEmailAddresses(a *cmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	if len(a.EmailSANs) <= 0 {
+		return nil
+	}
+	el := field.ErrorList{}
+	for i, d := range a.EmailSANs {
+		e, err := mail.ParseAddress(d)
+		if err != nil {
+			el = append(el, field.Invalid(fldPath.Child("emailSANs").Index(i), d, "invalid email address"))
+		} else if e.Address != d {
+			// Go accepts email names as per RFC 5322
+			// This checks if only the address is specified
```

Comment:
Does this mean you're checking whether it is prefixed with `mailto`? Could you expand the comment here, as well as the error message returned to the user to explain how they can fix their input if they've specified an email address incorrectly?

Reply:
clarified it further